cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ArtifactHttpHandlerTest.java

```diff
@@ -138,6 +138,33 @@ public void testPluginRequirements() throws Exception {
                         new Requirements(ImmutableSet.of(Table.TYPE)));
   }
 
+  @Test
+  public void testPluginInvalid() throws Exception {
+    // invalid plugin class json array which does not have a name
+    String pluginJSON = "[{\"type\":\"type\",\"description\":\"desc1\"," +
```

Comment:
This is really brittle, as any changes to the class will not be reflected here. It's better to create the actual object and pass it around.

Reply:
This represents an invalid JSON which can be sent from external call. Dont want to expose the default constructor of PluginClass here.

Reply:
You can still create the invalid object and test it. Using a string like this is not maintainable.

Reply:
Instead of doing all the null checks in the PluginClass constructor on fields that are not annotated as nullable, they should be done in the validate method. You can then create these invalid objects normally instead of dealing with strings in these tests